### PR TITLE
CLN: move mypy config to pyproject.toml

### PR DESCRIPTION
**Pull request by fangchenli**
xref #40656

### CODE REVIEW

File: pandas/tests/io/xml/test_xml.py [hunk 1]
```diff
@@ -682,7 +682,7 @@ def test_names_option_wrong_type(datapath, parser):
 
     with pytest.raises(TypeError, match=("is not a valid type for names")):
         read_xml(
-            filename, names="Col1, Col2, Col3", parser=parser  # type: ignore[arg-type]
```
Comment (fangchenli) [inline pandas/tests/io/xml/test_xml.py:685]: fixed by `namespace_packages = true`

Comment (simonjayhawkins) [reply pandas/tests/io/xml/test_xml.py:685]: fixed? Revealed type of `read_xml` is now "Any". without `namespace_packages = true` it is 

```
"def (path_or_buffer: Union[os.PathLike[builtins.str], builtins.str, typing.IO[Any], io.RawIOBase, io.BufferedIOBase, io.TextIOBase, io.TextIOWrapper, mmap.mmap], xpath: Union[builtins.str, None] =, namespaces: Union[builtins.dict[Any, Any], builtins.list[builtins.dict[Any, Any]], None] =, elems_only: Union[builtins.bool, None] =, attrs_only: Union[builtins.bool, None] =, names: Union[builtins.list[builtins.str], None] =, encoding: Union[builtins.str, None] =, parser: Union[builtins.str, None] =, stylesheet: Union[Union[os.PathLike[builtins.str], builtins.str, typing.IO[Any], io.RawIOBase, io.BufferedIOBase, io.TextIOBase, io.TextIOWrapper, mmap.mmap], None] =, compression: Union[builtins.str, builtins.dict[builtins.str, Any], None] =, storage_options: Union[builtins.dict[builtins.str, Any], None] =) -> pandas.core.frame.DataFrame"
```

**Review by simonjayhawkins — CHANGES_REQUESTED** [review]
Thanks @fangchenli for the PR. can you limit this one to just the move (to reflect the PR title). Can you open another PR for the configuration changes.

**Review by simonjayhawkins — APPROVED** [review]
Thanks @fangchenli lgtm pending green

**Pull request merged**